packages/graphback-datasync/src/helpers/createDataSync.ts

```diff
@@ -0,0 +1,27 @@
+import { GraphQLSchema } from 'graphql';
+import { Db } from 'mongodb';
+import { GraphbackAPIConfig, GraphbackAPI, buildGraphbackAPI, GraphbackConfig } from "graphback";
+import { PubSub } from 'graphql-subscriptions';
+import { createDataSyncMongoDbProvider, createDataSyncConflictProviderCreator } from '../providers';
+import { ConflictError, ConflictStateMap, DataSyncModelConflictConfig } from "../util";
+import { createDataSyncCRUDService } from '../services';
+import { DataSyncPlugin } from '../DataSyncPlugin';
+
+type DataSyncGraphbackConfig = Partial<GraphbackConfig>
+
+export function createDataSync(model: string | GraphQLSchema, db: Db, dataSyncConfig: { [modelName: string]: DataSyncModelConflictConfig} = {}, graphbackConfig: DataSyncGraphbackConfig = {}): GraphbackAPI {
+
+  return buildGraphbackAPI(model, {
+    serviceCreator: createDataSyncCRUDService({ pubSub: new PubSub() }),
+    ...graphbackConfig,
+    dataProviderCreator: createDataSyncConflictProviderCreator(db, dataSyncConfig),
+    plugins:[
+      ...graphbackConfig.plugins || [],
+      new DataSyncPlugin({ modelUsesVersion: Object.keys(dataSyncConfig).reduce((versionMap: any, modelName: string) => {
```

Comment:
Separate variable?

Reply:
Actually why we do not do this inside plugin. It feels wrong to do it here.

Reply:
+1 it should work the same if a user wants to use `buildGraphbackAPI` directly instead of `createDataSync`